src/internal_events/parser.rs

```diff
@@ -0,0 +1,122 @@
+use metrics::counter;
+use vector_core::internal_event::InternalEvent;
+
+#[derive(Debug)]
+pub struct ParserMatchError<'a> {
+    pub value: &'a [u8],
+}
+
+impl InternalEvent for ParserMatchError<'_> {
+    fn emit_logs(&self) {
+        error!(
+            message = "Pattern failed to match.",
+            error = "Failed to match pattern",
+            error_type = "condition_failed",
+            stage = "processing",
+            field = &super::truncate_string_at(&String::from_utf8_lossy(self.value), 60)[..],
+            internal_log_rate_secs = 30
+        );
+    }
+
+    fn emit_metrics(&self) {
+        counter!(
+            "component_errors_total", 1,
+            "error" => "Failed to match pattern",
+            "error_type" => "condition_failed",
+            "stage" => "processing",
+        );
+        // deprecated
+        counter!("processing_errors_total", 1, "error_type" => "failed_match");
+    }
+}
+
+#[derive(Debug)]
+pub struct ParserMissingFieldError<'a> {
+    pub field: &'a str,
+}
+
+impl InternalEvent for ParserMissingFieldError<'_> {
+    fn emit_logs(&self) {
+        error!(
+            message = "Field does not exist.",
+            field = %self.field,
+            error = "Field not found",
+            error_type = "condition_failed",
+            stage = "processing",
+            internal_log_rate_secs = 10
+        );
+    }
+
+    fn emit_metrics(&self) {
+        counter!(
+            "component_errors_total", 1,
+            "error" => "Field not found",
+            "error_type" => "condition_failed",
+            "stage" => "processing",
+            "field" => self.field.to_string(),
+        );
+        // deprecated
+        counter!("processing_errors_total", 1, "error_type" => "missing_field");
+    }
+}
+
+#[derive(Debug)]
+pub struct ParserTargetExistsError<'a> {
+    pub target_field: &'a str,
+}
+
+impl<'a> InternalEvent for ParserTargetExistsError<'a> {
+    fn emit_logs(&self) {
+        error!(
+            message = "Target field already exists.",
```

Comment:
I agree with you but I have a lack of inspiration 😅

Reply:
correct me if I'm wrong but the `message` field should be something more personalized while `error` would be something we could filter on in a Datadog interface for example?
If I follow this logic, then `error` should have something like `Target field already exists` because there is a tag `field` as well to help filtering. So message should be something like `format!("Target field {:?} already exists", self.target_field)` so that we can have a representation of the error?

Reply:
Yep, I think `message` is meant to be descriptive for users while `error` might not always be (e.g. a status code). That suggestion sounds good to me.

Reply:
Yeah, agreed here, `error` should be closer to an error code -- one of a static set of values -- while `message` should be meant for human consumption.